src/Workspaces/Core/Portable/FindSymbols/FindReferences/FindReferenceCache.cs

```diff
@@ -100,19 +102,20 @@ public static SymbolInfo GetSymbolInfo(SemanticModel model, SyntaxNode node, Can
                 {
                     if (SyntaxTreeIdentifierInfo.TryGetIdentifierLocations(document, version, text, positions.Object, cancellationToken))
                     {
-                        return GetTokensFromText(root, positions.Object, text, candidate, cancellationToken).ToList();
+                        return GetTokensFromText(root, positions.Object, text, candidate, cancellationToken);
                     }
                 }
 
-                return GetTokensFromText(syntaxFacts, root, content, text, candidate, cancellationToken).ToList();
+                return GetTokensFromText(syntaxFacts, root, content, text, candidate, cancellationToken);
             }
 
-            return SpecializedCollections.EmptyList<SyntaxToken>();
+            return ImmutableArray<SyntaxToken>.Empty;
         }
 
-        private static IEnumerable<SyntaxToken> GetTokensFromText(
+        private static ImmutableArray<SyntaxToken> GetTokensFromText(
```

Comment:
This method used to lazily return the `SyntaxToken`s one at a time; now it stuffs them all into an array upfront. It seems like this is only a win if the callers ended up doing that anyway. Is that the case?